source/configuration/modules/omfile.rst

```diff
@@ -1,434 +1,437 @@
-
-.. function::  zipLevel [level]
-
-   *Default: 0*
-
-   if greater 0, turns on gzip compression of the output file. The
-   higher the number, the better the compression, but also the more CPU
-   is required for zipping.
-
-.. function::  veryRobustZip [switch]
-
-   *Default: on*
+Apart from generating invalid paths/filenames, the most common cause of this is permission problems.
+Permission problems can be either traditional filesystem permissions (the files need to be able to be created after any privilege drop takes place), or they can be a security module like SELinux or AppArmor.
+
+==================  ==================
+dynaFileCacheSize:  "number"
+Default:            "10"
+Legacy:             $DynaFileCacheSize
+==================  ==================
+
+**Applies only if dynamic filenames are used.**
+
+Specifies the number of files that will be kept open.
+Note that this is a per-action parameter, so if you have
+multiple dynafile actions, each of them have their individual caches
+(which means the numbers sum up).
+
+Ideally, the cache size exactly
+matches the need. You can use :doc:`impstats <impstats>` to tune
+this value.
+Note that a *lower-than-needed* cache size may be a lot worse that a *bigger-than-needed*.
+
+=========  =============================================
+template:  "template-name"
+Default:   same as module parameter (RSYSLOG_FileFormat)
+Legacy:    use: ". /var/log/foo;template-name"
+=========  =============================================
+
+Sets the template to be used for this action.
+
+================  ==============================
+closeTimeout:     "number"
+Default:          "0 if using *file*, 10 if using *dynaFile*"
+Available since:  8.3.3
+================  ==============================
+
+Specifies after how many minutes of inactivity a file is
+automatically closed.
+
+Note that this functionality is implemented based on the  :doc:`janitor process <../../concepts/janitor>`.
+
+=========  ===============
+zipLevel:  "number"
+Default:   "0"
+Legacy:    $OMFileZipLevel
+=========  ===============
+
+If greater than 0, turns on gzip compression of the output file. The
+higher the number, the better the compression, but also the more CPU
+is required for zipping.
+
+================  ========
+veryRobustZip:    "on|off"
+Default:          "off"
+Available since:  7.3.0
+================  ========
+
+If *zipLevel* is greater than 0,
+then this setting controls if extra headers are written to make the
+resulting file extra hardened against malfunction. 
+
+If set to *off*, data appended to previously unclean closed files may not be
+accessible without extra tools, something usually bearable.
+On the other hand, if set to *on*, the extra headers considerably degrade compression, resulting in files four to five times bigger.
+
+Filesystem parameters
+---------------------
+
+=========  ================
+dirOwner:  "username"
+Default:   *system default*
+Legacy:    $DirOwner
```

Comment:
consider adding a note to this that says that the default can be set in the module load